lib/apiv2/felixconfig.go

```diff
@@ -0,0 +1,152 @@
+	IpsetsRefreshInterval              *time.Duration `json:"ipsetsRefreshInterval,omitempty"`
+	MaxIpsetSize                       *int           `json:"maxIpsetSize,omitempty"`
+
+	NetlinkTimeoutSecs *time.Duration `json:"netlinkTimeoutSecs,omitempty"`
+
+	MetadataAddr string `json:"metadataAddr,omitempty"`
+	MetadataPort *int   `json:"metadataPort,omitempty"`
+
+	InterfacePrefix string `json:"interfacePrefix,omitempty"`
+
+	ChainInsertMode             string `json:"chainInstertMode,omitempty"`
+	DefaultEndpointToHostAction string `json:"defaultEndpointToHostAction,omitempty"`
+	IptablesFilterAllowAction   string `json:"iptablesFilterAllowAction,omitempty"`
+	IptablesMangleAllowAction   string `json:"iptablesMangleAllowAction,omitempty"`
+	LogPrefix                   string `json:"logPrefix,omitempty"`
+
+	LogFilePath string `json:"logFilePath,omitempty"`
+
+	LogSeverityFile   string `json:"logSeverityFile,omitempty"`
+	LogSeverityScreen string `json:"logSeverityScreen,omitempty"`
+	LogSeveritySys    string `json:"logSeveritySys,omitempty"`
+
+	IpInIpEnabled *bool `json:"ipInIpEnabled,omitempty"`
+	IpInIpMtu     *int  `json:"ipInIpMtu,omitempty"`
+
+	ReportingIntervalSecs *time.Duration `json:"reportingIntervalSecs,omitempty"`
+	ReportingTTLSecs      *time.Duration `json:"reportingTTLSecs,omitempty"`
+
+	EndpointReportingEnabled   *bool          `json:"endpointReportingEnabled,omitempty"`
+	EndpointReportingDelaySecs *time.Duration `json:"endpointReportingDelay,omitempty"`
+
+	IptablesMarkMask *uint32 `json:"iptablesMarkMask,omitempty"`
+
+	DisableConntrackInvalidCheck *bool `json:"disableConntrackInvalidCheck,omitempty"`
+
+	HealthEnabled                   *bool `json:"healthEnabled,omitempty"`
+	HealthPort                      *int  `json:"healthPort,omitempty"`
+	PrometheusMetricsEnabled        *bool `json:"prometheusMetricsEnabled,omitempty"`
+	PrometheusMetricsPort           *int  `json:"prometheusMetricsPort,omitempty"`
+	PrometheusGoMetricsEnabled      *bool `json:"prometheusGoMetricsEnabled,omitempty"`
+	PrometheusProcessMetricsEnabled *bool `json:"prometheusProcessMetricsEnabled,omitempty"`
+
+	FailsafeInboundHostPorts  *[]ProtoPort `json:"failsafeInboundHostPorts,omitempty"`
+	FailsafeOutboundHostPorts *[]ProtoPort `json:"failsafeOutboundHostPorts,omitempty"`
+
+	UsageReportingEnabled *bool `json:"usageReportingEnabled,omitempty"`
+	// TODO: Check if CalicoVersion is important
+	CalicoVersion string `json:"calicoVersion,omitempty"`
+
+	DebugMemoryProfilePath          string         `json:"debugMemoryProfilePath,omitempty"`
+	DebugDisableLogDropping         *bool          `json:"debugDisableLogDropping,omitempty"`
+	DebugSimulateCalcGraphHangAfter *time.Duration `json:"debugSimulateCalcGraphHangAfter,omitempty"`
+	DebugSimulateDataplaneHangAfter *time.Duration `json:"debugSimulateDataplaneHangafter,omitempty"`
+}
+
+// Define the ProtoPort here. Cannot use the equivalent ProtoPort in felix since
```

Comment:
Don't think you need the circular dependencies comment - generally it doesn't make sense for anything in libcalico-go to import things from felix because felix should be importing from here.